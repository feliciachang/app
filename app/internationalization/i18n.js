import RNLanguages from "react-native-languages";
import i18n from "i18n-js";

import en from "./translations/en.json";
import es from "./translations/es.json";
import pt from "./translations/pt.json";
import fr from "./translations/fr.json";

i18n.locale = RNLanguages.language;
console.log(i18n.locale);
i18n.fallbacks = true;
i18n.translations = { en, es, pt, fr };

export default i18n;
