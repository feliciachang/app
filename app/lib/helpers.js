'use strict';

export function unixNow() {
    return Math.round((new Date()).getTime() / 1000);
}
