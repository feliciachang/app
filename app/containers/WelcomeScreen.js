import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { View, Image, ScrollView } from "react-native";

import RNLanguages from "react-native-languages";
import i18n from "../internationalization/i18n";

import { AppScreen, MenuButtonContainer, MenuButton } from "../components";

import { initialize, navigateConnecting, browseDirectory, navigateBrowser, navigateEasyModeWelcome, navigateAbout, navigateMap, deviceStartConnect, deviceStopConnect } from "../actions";

import styles from "../styles";

class WelcomeScreen extends React.Component {
    static navigationOptions = ({ navigation }) => {
        return { title: i18n.t("welcome.title") };
    };

    componentDidMount() {
        this.props.initialize();
        this.props.deviceStartConnect();
    }

    render() {
        const { navigateConnecting, browseDirectory, navigateEasyModeWelcome, navigateMap, navigateAbout } = this.props;

        return (
            <ScrollView>
                <AppScreen>
                    <Image
                        source={require("../../assets/fk-header.png")}
                        style={{
                            resizeMode: "contain",
                            width: "100%",
                            height: 200
                        }}
                    />
                    <MenuButtonContainer>
                        <MenuButton title={i18n.t("welcome.connect")} onPress={() => navigateConnecting()} />
                        <MenuButton title={i18n.t("welcome.browser")} onPress={() => browseDirectory("/")} />
                        <MenuButton title={i18n.t("welcome.mode")} onPress={() => navigateEasyModeWelcome()} />
                        <MenuButton title={i18n.t("welcome.map")} onPress={() => navigateMap()} />
                        <MenuButton title={i18n.t("welcome.about")} onPress={() => navigateAbout()} />
                    </MenuButtonContainer>
                </AppScreen>
            </ScrollView>
        );
    }
}

WelcomeScreen.propTypes = {
    initialize: PropTypes.func.isRequired,
    navigateConnecting: PropTypes.func.isRequired,
    browseDirectory: PropTypes.func.isRequired,
    navigateEasyModeWelcome: PropTypes.func.isRequired,
    navigateAbout: PropTypes.func.isRequired,
    navigateMap: PropTypes.func.isRequired,
    deviceStartConnect: PropTypes.func.isRequired
};

const mapStateToProps = state => ({});

export default connect(
    mapStateToProps,
    {
        initialize,
        navigateConnecting,
        navigateEasyModeWelcome,
        navigateMap,
        navigateAbout,
        browseDirectory,
        deviceStartConnect
    }
)(WelcomeScreen);
