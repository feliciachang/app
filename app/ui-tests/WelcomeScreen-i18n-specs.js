import React from "react";
import WelcomeScreen from "../containers/WelcomeScreen";
import { mockFunctionHelper } from "../../specs/helpers";

import { shallow, mount, render } from "enzyme";
import i18n from "../internationalization/i18n";
import toJson from "enzyme-to-json";

let store = mockFunctionHelper({});

describe("<WelcomeScreen/>", () => {
  describe("render()", () => {
    test("renders the component", () => {
      i18n.locale = "es-US";
      const wrapper = shallow(<WelcomeScreen store={store} />);
      const component = wrapper.dive();
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
