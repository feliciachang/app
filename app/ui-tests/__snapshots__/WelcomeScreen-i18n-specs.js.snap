// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeScreen/> render() renders the component 1`] = `
<ScrollView>
  <AppScreen>
    <Component
      source={
        Object {
          "testUri": "../../../assets/fk-header.png",
        }
      }
      style={
        Object {
          "height": 200,
          "resizeMode": "contain",
          "width": "100%",
        }
      }
    />
    <MenuButtonContainer>
      <MenuButton
        onPress={[Function]}
        title="Conectar"
      />
      <MenuButton
        onPress={[Function]}
        title="Navegador"
      />
      <MenuButton
        onPress={[Function]}
        title="Modo FÃ¡cil"
      />
      <MenuButton
        onPress={[Function]}
        title="Map"
      />
      <MenuButton
        onPress={[Function]}
        title="Sobre"
      />
    </MenuButtonContainer>
  </AppScreen>
</ScrollView>
`;
